R/tab-book.R

```diff
@@ -62,15 +65,29 @@ tabBook <- function(multitable, dataset, weight = crunch::weight(dataset),
         fmt <- match.arg(output_format)
     }
 
-    accept <- extToContentType(fmt)
-    if (missing(file)) {
-        if (fmt == "json") {
-            ## We don't need a file.
-            file <- NULL
-        } else {
-            ## Generate a reasonable filename in the current working dir
-            file <- paste(name(multitable), fmt, sep = ".")
-        }
+    if (is.null(weight) | is.variable(weight)) {
+        tabBookSingle(multitable, dataset, weight, fmt, file, filter, use_legacy_endpoint, dots)
+    } else if (is.list(weight)) {
+        tabBookMulti(multitable, dataset, weight, fmt, file, filter,  use_legacy_endpoint, dots)
+    } else {
+        stop("weight must be NULL, a CrunchVariable or a list indicating a multi-weight spec")
```

Comment:
Based on the updated tabWeightSpec it could also be a data.frame

Reply:
Yeah, `is.list()` returns `TRUE` for `data.frame`s, but I added a check for data.frame so that the code is clearer